### DISCUSSION TITLE
Sign Handler

### DISCUSSION THREAD

**Discussion by Rqver** [Minestom/Minestom#661] [Help] [answered]
Hi,

I am trying to create a sign handler to load in anvil worlds, @emortaldev was very helpful in response to my last question (#657).

I have the below sign handler, and I am trying to register the handler using
```java
BlockHandler signHandler = new SignHandler();
MinecraftServer.getBlockManager().registerHandler("minecraft:sign", signHandler);
```
However it errors: `Cannot resolve method 'registerHandler(java.lang.String, net.minestom.server.instance.block.BlockHandler)'`.
I also tried replacing `"minecraft:sign"` with `NamespaceID.from("minecraft:sign")`, but had the same issue.

Sign Handler:
```java
public class SignHandler implements BlockHandler {
    @Override
    public @NotNull Collection<Tag<?>> getBlockEntityTags() {
        return List.of(Tag.Byte("GlowingText"), Tag.String("Color"), Tag.String("Text1"), Tag.String("Text2"), Tag.String("Text3"), Tag.String("Text4"));
    }

    @Override
    public @NotNull NamespaceID getNamespaceId() {
        return NamespaceID.from(Key.key("minecraft:sign"));
    }
}
```

Thanks

**Comment by emortaldev** [comment, marked as answer]
I think it's something like `MinecraftServer.getBlockManager().registerHandler("minecraft:sign", () -> signHandler);`
(It is type Supplier)

**Reply by Rqver** [reply]
That didnt work - signHandler isn't a statement and needs to be apparently. 
Also tried replacing signHandler with `new SignHandler();`, but:
`Cannot resolve method 'registerHandler(java.lang.String, <lambda expression>)'`

**Reply by emortaldev** [reply]
Try with just `() -> signHandler`?

**Reply by Rqver** [reply]
Omg that works

And I only just noticed thats what you originally said. 
Srry - its been a long day.

**Reply by emortaldev** [reply]
No it was an edit incase it was the actual answer 😛

**Reply by CraziiDust** [reply]
You can replace lambda expression with `SignHandler::new`

**Comment by ch99q** [comment]
Looking at this, the block handler could be a great way to implement an i18n localization solution for signs. Is there any way trough this method to know what block it is through https://javadoc.minestom.net/net/minestom/server/instance/block/BlockHandler.html#getBlockEntityAction()?
